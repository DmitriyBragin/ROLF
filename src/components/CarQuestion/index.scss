.car-question {
    display: flex;
    flex-direction: column;
    margin-bottom: 40px;
    &__header {
        display: flex;
        flex-direction: row;
        font-weight: 700;
        &_name {
        }
        &_car {
            margin-left: 20px;
            color: #923562;
        }
    }
    &__time {
        color: grey;
        margin-bottom: 10px;
    }
    &__text {
    }
}
