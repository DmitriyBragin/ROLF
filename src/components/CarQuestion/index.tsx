import './index.scss';
import { Question } from '../../utils/types';

export const CarQuestion = ({ question }: { question: Question }) => {
    const transformDate = (): string => {
        if (!question.date) return 'N/A';
        return question.date.split('T').join(' Ð² ');
    };

    return question ? (
        <div className="car-question">
            <div className="car-question__header">
                <div className="car-question__header_name">{question.user}</div>
                <div className="car-question__header_car">
                    {question.carBrand} {question.carModel}
                </div>
            </div>
            <div className="car-question__time">{transformDate()}</div>
            <div className="car-question__text">{question.text}</div>
        </div>
    ) : (
        <></>
    );
};
