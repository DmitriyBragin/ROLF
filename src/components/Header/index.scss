.header {
    color: #3f48cc;
    height: 75px;
    display: flex;
    flex-direction: row;
    flex: 0 0 75px;
    align-items: center;
    font-size: 24px;
    font-weight: 700;
}
