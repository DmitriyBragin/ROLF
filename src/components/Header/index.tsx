import './index.scss';

export const Header = () => {
    return <div className="header">Вопросы по автомобилям</div>
}