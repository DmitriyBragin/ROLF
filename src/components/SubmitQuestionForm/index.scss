.submit-question-form {
    width: 50%;
    @media screen and (max-width: 500px) {
        width: 100%;
    }
}

.MuiFormHelperText-root {
    text-align: right !important;
    margin: 0 !important;
}
