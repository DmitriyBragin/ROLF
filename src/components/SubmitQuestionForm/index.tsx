import './index.scss';
import { Question } from '../../utils/types';
import { Box, Button, Grid, TextField } from '@mui/material';
import React, { useEffect, useState } from 'react';
import { getEmptyQuestionObject, isFilled, preparePayload, validateForm } from './helpers';

export const SubmitQuestionForm = ({ cb }: { cb: (q: Question) => void }) => {
    const [formErrors, setFormErrors] = useState<Question>(getEmptyQuestionObject());
    const [data, setData] = useState<Question>(getEmptyQuestionObject());
    const [canSubmit, setCanSubmit] = useState<boolean>(false);

    const handleChange = (evt: { target: any }) => {
        setData({
            ...data,
            [evt.target.name]: evt.target.value,
        });
    };

    useEffect(() => {
        setCanSubmit(isFilled(data));
    }, [JSON.stringify(data)]);

    const cleanupData = (): void => {
        setData(getEmptyQuestionObject());
        setFormErrors(getEmptyQuestionObject());
    };

    const handleSubmit = (e: any) => {
        e.preventDefault();
        const { isValid, errors } = validateForm(data);
        setFormErrors(errors);
        if (isValid) {
            const payload: Question = preparePayload(data);
            cleanupData();
            cb(payload);
        }
    };

    return (
        <div className="submit-question-form">
            <Box component="form" onSubmit={handleSubmit} sx={{ mt: 3 }}>
                <Grid container spacing={2}>
                    <Grid item xs={12} sm={6}>
                        <TextField
                            name="user"
                            required
                            fullWidth
                            id="user"
                            label="Имя"
                            value={data.user}
                            onChange={handleChange}
                            error={!!formErrors.user.length}
                            helperText={formErrors.user}
                        />
                    </Grid>
                    <Grid item xs={12} sm={6}>
                        <TextField
                            required
                            fullWidth
                            id="email"
                            label="Email"
                            name="email"
                            value={data.email}
                            error={!!formErrors.email.length}
                            helperText={formErrors.email}
                            onChange={handleChange}
                        />
                    </Grid>
                    <Grid item xs={12} sm={6}>
                        <TextField
                            required
                            fullWidth
                            id="carBrand"
                            label="Марка авто"
                            name="carBrand"
                            value={data.carBrand}
                            error={!!formErrors.carBrand.length}
                            helperText={formErrors.carBrand}
                            onChange={handleChange}
                        />
                    </Grid>
                    <Grid item xs={12} sm={6}>
                        <TextField
                            required
                            fullWidth
                            id="carModel"
                            label="Модель авто"
                            name="carModel"
                            value={data.carModel}
                            error={!!formErrors.carModel.length}
                            helperText={formErrors.carModel}
                            onChange={handleChange}
                        />
                    </Grid>
                    <Grid item xs={12}>
                        <TextField
                            multiline
                            required
                            fullWidth
                            name="text"
                            label="Введите текст вопроса"
                            type="text"
                            id="text"
                            value={data.text}
                            inputProps={{ maxLength: 500 }}
                            error={!!formErrors.text.length}
                            helperText={`${formErrors.text} ${data.text.length}/500`}
                            onChange={handleChange}
                        />
                    </Grid>
                </Grid>
                <Button
                    type="submit"
                    variant="contained"
                    sx={{ mt: 3, mb: 2 }}
                    disabled={!canSubmit}>
                    Отправить
                </Button>
            </Box>
        </div>
    );
};
