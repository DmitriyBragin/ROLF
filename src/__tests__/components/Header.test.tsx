import { render } from '@testing-library/react';
import React from 'react';

import { Header } from '../../components/Header';

describe('Header', () => {
    test('renders', () => {
        const { container } = render(<Header />);
        expect(container).toMatchSnapshot();
    });
});
