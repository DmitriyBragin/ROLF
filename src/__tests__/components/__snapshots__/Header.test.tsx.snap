// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div>
  <div
    class="header"
  >
    Вопросы по автомобилям
  </div>
</div>
`;
